src/main/scala/entities/people/Student.scala

```diff
@@ -4,17 +4,25 @@ import java.util.regex.Pattern
 
 import entities.course.{Course, CourseRole}
 import entities.module.{Module, ModuleRole}
+import exceptions.InvalidJsonException
 import services.Utils
 import services.generator.Generator
 import services.parser.TimeTableParser
-
+import services.JsonUtils._
 import scala.collection.mutable
 import scala.collection.mutable.ListBuffer
 import scala.io.Source
 import scala.util.Random
 
 
 object Student extends Generator[Student] {
+  def apply(json:Map[String, Any]): Student = {
+    if(json("studentId") == null) throw new InvalidJsonException("Missing Id")
```

Comment:
Surely if we don't provide an ID when adding a student we want one to be automatically generated?

Reply:
Yeah okay I'll change this